src/app/shared/pipes/itemRoute.ts

```diff
@@ -26,14 +26,14 @@ export class ItemRouteWithAnswerPipe implements PipeTransform {
 
 @Pipe({ name: 'withParentAttempt', pure: true })
 export class ItemRouteWithParentAttemptPipe implements PipeTransform {
-  transform<T extends RawItemRoute>(route: T, parentAttemptId: ItemRoute['parentAttemptId']): T {
+  transform<T extends RawItemRoute>(route: T, parentAttemptId: AttemptId): T {
     return { ...route, parentAttemptId };
   }
 }
 
 @Pipe({ name: 'withPath', pure: true })
 export class ItemRouteWithPathPipe implements PipeTransform {
-  transform<T extends RawItemRoute>(route: T, path: string[], root?: string): T {
+  transform<T extends RawItemRoute>(route: T, path: string[], root?: string): ItemRoute {
```

Comment:
That's not correct as a `FullItemRoute` would become a `ItemRoute` (so would "downgrade").
That's why I was writing :
> a RawItemRoute would with a path would become a ItemRoute but can we express that in TS?